components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -1346,9 +1348,22 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
                     return { existingWorkspaces: runningForContext };
                 }
             }
-            const project = CommitContext.is(context)
-                ? (await this.projectsService.findProjectsByCloneUrl(user.id, context.repository.cloneUrl))[0]
-                : undefined;
+
+            let project: Project | undefined = undefined;
+            if (options.projectId) {
+                project = await this.projectsService.getProject(user.id, options.projectId);
+            } else if (CommitContext.is(context)) {
+                const projects = await this.projectsService.findProjectsByCloneUrl(
+                    user.id,
+                    context.repository.cloneUrl,
+                );
+                if (projects.length > 1) {
```

Comment:
if the user didn't pass in a projectId but we find more than one candidate we throw and need to tell the user.